Assets/LeapMotion/Core/Scripts/DataStructures/Pool.cs

```diff
@@ -85,7 +83,29 @@ public static class Pool<T> where T : new() {
       return value;
     }
 
+    private static T spawnValue() {
+      T value;
+      int count = 0;
+      do {
+        if (_pool.Count > 0) {
+          value = _pool.Pop();
+        }
+        else {
+          value = new T();
+        }
+        count++;
+      } while (value == null && count < 100);
+      if (count == 100) {
```

Comment:
Unless new T() returns null. Again, unless that's a strict impossibility, I'm concerned about having any code anywhere that could hard-lock the editor, even in the wildest scenarios... I won't push too hard for it though. I can remove the counter if you'd prefer to do the check in Recycle